app/code/core/Mage/Index/Model/Event.php

```diff
@@ -171,6 +171,8 @@ protected function _mergeNewDataRecursive($previous, $current)
                 }
             } elseif (!array_key_exists($key, $current) || is_null($current[$key])) {
                 $current[$key] = $previous[$key];
+            } elseif (array_key_exists($key, $current) && array_key_exists($key, $previous) && ($previous[$key] == $current[$key])) {
+                return $current;
```

Comment:
Since this is inside a foreach loop it should not `return`  in this case. Take for example:

```
$previous = ["a", "b"];
$current = ["a", "c"];
```

My understanding is the result should ideally be `["a", "b", "c"]` but this patch would yield `["a", "c"]` (with the current code probably yielding `["a", "b", "a", "c"]`).

It may be better to allow a duplicated value and then address that using `array_unique` like your other commit does since it is hard to say what making changes to this method could impact.

Reply:
that's why i did not dare to edit properly this function as it has a broader area of impact, do you want for me to revert these changes ?

Reply:
Unless you have a lot of time to spend improving the method then that would probably be the best way forward. While the old behavior is not optimal it can still yield a correct outcome given your other patch using array_unique, from what I understand.

I'd love to know more about what issue this PR fixes though.

Reply:
current spot is in regard with reindexing category url which added duplication thus making extra cycle for already parsed categories.

As for other, will try to take a look and see if i can improve core function, which i am scared to add a fix for :fearful: